### PR TITLE
fix(ground_segmentation): fix  non_ground segmentation missing

### PR DESCRIPTION
**Pull request by badai-nguyen**
Signed-off-by: badai-nguyen <dai.nguyen@tier4.jp>

## Description

Scan_ground_filter misses some parts of non_ground segments in some cases like objects close to each other. 

image

## Tests performed

image

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by yukkysaito** [comment]
lools good.
@miursh can you check the algorithm?

File: perception/ground_segmentation/src/scan_ground_filter_nodelet.cpp [hunk 1]
```diff
@@ -166,8 +166,11 @@ void ScanGroundFilterComponent::classifyPointCloud(
         (points_distance <
          (p->radius * radial_divider_angle_rad_ + split_points_distance_tolerance_));
 
+      float global_slope = std::atan2(p->orig_point->z, p->radius);
       // check points which is far enough from previous point
-      if (
+      if (global_slope > global_slope_max_angle) {
+        p->point_state = PointLabel::NON_GROUND;
```
Comment (miursh) [inline perception/ground_segmentation/src/scan_ground_filter_nodelet.cpp:172]: [IMO] Could you add `        calculate_slope = false;` this to the line below? Just for an explicit expression.

**Review by miursh — COMMENTED** [review]
Just one comment.

File: perception/ground_segmentation/src/scan_ground_filter_nodelet.cpp (hunk 1 above)
Comment (badai-nguyen) [reply perception/ground_segmentation/src/scan_ground_filter_nodelet.cpp:172]: Thanks for your comment. I have add it.

**Review by miursh — APPROVED** [review]
LGTM

**Pull request merged**